concurrent_test.go

```diff
@@ -116,24 +119,19 @@ func concurrentReadAndWrite(t *testing.T,
 
 	stopCh := make(chan struct{}, 1)
 	errCh := make(chan error, readerCount+1)
+	recordingCh := make(chan historyRecord, readerCount+1)
 
-	// start readonly transactions
-	g := new(errgroup.Group)
-	for i := 0; i < readerCount; i++ {
-		reader := &readWorker{
-			db:              db,
-			bucket:          bucket,
-			keys:            keys,
-			minReadInterval: minReadInterval,
-			maxReadInterval: maxReadInterval,
-			errCh:           errCh,
-			stopCh:          stopCh,
-			t:               t,
-		}
-		g.Go(reader.run)
-	}
+	// collect history records
+	var records historyRecords
+	var wg sync.WaitGroup
```

Comment:
having two sets of async goroutines synchronized on different primitives is hard to follow. Maybe move workers to sub function?

Reply:
Makes sense, let me enhance it in a followup PR.

Reply:
done